logstash-core/src/main/java/org/logstash/ackedqueue/Page.java

```diff
@@ -79,6 +80,27 @@ public void write(byte[] bytes, long seqNum, int checkpointMaxWrites) throws IOE
         }
     }
 
+    public void writeMany(List<byte[]> byteList, long seqNum, int checkpointMaxWrites) throws IOException {
```

Comment:
@andrewvc I think it would be good to consumer `List<Queueable>` here instead of `List<byte[]>`. Tbh. any API using `byte[]` in a serialize then write setting is kinda horrible. Let's not add more of them. See #7039 for some more on this.

I'd just do the serialization in line here => have the `byte[]` live for a shorter time + stay more flexible in terms of fixing our serialization eventually :)

Reply:
@andrewvc nevermind I see now that we do the size check upstream ... urgh :( but can't fix this here ... dropping above point